### ISSUE TITLE
Problem at compile time with latest Android Studio 3.5.1

### ISSUE THREAD

**Issue by nicolasjafelle** [muxinc/mux-stats-sdk-exoplayer#19]
- Cannot compile project after update to latest gradle plugin `3.5.1` and gradle wrapper `5.6.2-all`
- Simple update to latest `Android Studio 3.5.1` and update gradle plugin to `3.5.1` and gradle wrapper `5.6.2-all` and try to rebuild project.
- ExoPlayer version `2.9.6` with Custom Integration based on the demo project.
- There is no bug report and the problem is in compilation time. Might be a gradle incompatibility.
- Works with `gradle plugin 3.4.2` and `gradle wrapper 5.1.1-all`

Output
```Caused by: com.android.tools.r8.utils.AbortException: Error: Default method desugaring of `com.mux.stats.sdk.muxstats.MuxStatsExoPlayer` failed because it's hierarchy is incomplete. The class `com.mux.stats.sdk.core.events.EventBus` is missing and it is the declared super class of `com.mux.stats.sdk.muxstats.MuxBaseExoPlayer` ```

**Comment by jsanford8** [comment]
fixed by #20

**Comment by nicolasjafelle** [comment]
@jsanford8 thanks for fixing the issue. Unfortunately I still have it, do you remember what do the trick?

**Comment by jsanford8** [comment]
hey @nicolasjafelle - have you pulled the lastest version and run a gradle sync? I'm using Android Studio 3.5.1 just with what's currently in master, and it works for me (everything that we changed should be checked in appropriately)

**Comment by nicolasjafelle** [comment]
yes! quick question: it seems you are using `Android SDK Build-Tools 28.0.3` and I am using `29.0.2`
Do you think that maybe that is the problem?

**Comment by jsanford8** [comment]
Hmm, that could potentially be the issue. I can try to change that and see what happens on my side, one minute

**Comment by nicolasjafelle** [comment]
just tested to use `28.0.3` and my problem still happens...

**Comment by jsanford8** [comment]
Have you tried running Build > Clean Project since you pulled the latest commits?

**Comment by nicolasjafelle** [comment]
BTW I also face this:

```
AGPBI: {"kind":"error","text":"Default method desugaring of `com.mux.stats.sdk.muxstats.MuxStatsExoPlayer` failed because it's hierarchy is incomplete. The class `com.mux.stats.sdk.core.events.EventBus` is missing and it is the declared super class of `com.mux.stats.sdk.muxstats.MuxBaseExoPlayer`","sources":[{}],"tool":"D8"}
```

Do you think I need to add a proguard rule like:

`-keep public class com.mux.stats.sdk.core.events.EventBus { public protected *; }`

**Comment by jsanford8** [comment]
Do you have the MuxCore.jar library correctly attached to your project? That looks like it can't link correctly to something in the MuxCore library.

**Comment by nicolasjafelle** [comment]
Yes yes, the funny thing is that I only receive this error if I go to `Build` -> `Rebuild project`

But if I can just run my app it works correct, and in fact I am seeing data in Mux Web console.

**Comment by jsanford8** [comment]
That's quite odd, honestly. I just re-ran it myself off of master, and things work. You mentioned that you have your own custom integration (for 2.9.6, which we didn't support until just now); any chance I can get you to clone the repo fresh and try to build the demo app as is to see if you still have the same problem?

**Comment by nicolasjafelle** [comment]
yes! I think so, let me do a small project with my Mux implementation and go back to you. BTW I am using d8 not proguard, are you the same? Thanks!

**Comment by jsanford8** [comment]
We're just using proguard standard, but I wonder if that might be part of the difference. I'm not actually familiar with D8; do you have a link to it so I can try to set it up on our side?

**Comment by nicolasjafelle** [comment]
in Android Studio 3.5.1 you just need to add `android.enableD8=true` in your gradle.properties file and that's it.

Check this: https://jakewharton.com/r8-optimization-staticization/

**Comment by jsanford8** [comment]
Thanks! I turned that on, and Build > Clean followed by Build > Rebuild Project worked without any issues for me

**Comment by nicolasjafelle** [comment]
So it must be something from my side then...

**Comment by nicolasjafelle** [comment]
@jsanford8 found the fix for me, I had to upgrade the `minSdkVersion 24` and works perfectly. Unfortunately for me I have to suppor Android 21. 

Do you know why upgrading to 24 is the minimum using ExoPlayer 2.9.6 ??

**Comment by jsanford8** [comment]
Hey @nicolasjafelle - according to the [docs from ExoPlayer 2.9.6](https://github.com/google/ExoPlayer/tree/r2.9.6), this requires Java 1.8 support, which is only supported from min API version 24. If you need to support API version 21, you might need to stick with ExoPlayer 2.8.x

**Comment by nicolasjafelle** [comment]
@jsanford8 I am not sure about it. I Think it only needs to add:
```
compileOptions {
  targetCompatibility JavaVersion.VERSION_1_8
}
```
In your app's build.gradle file and it just works. In fact the minSdkVersion across all the ExoPlayer repo is 16: [ExoPlayer Constants definition](https://github.com/google/ExoPlayer/blob/d73d64bc6fc90aa972c8e9f76fafba1ec661ca2d/constants.gradle)

Apart from that we have been using Exoplayer 2.9.6 with Mux for almost one year with android studio 3.4 and everything works out of the box with no problem. We started gettings this error while we updated to AS 3.5.
